src/pages/journal/journal.effects.ts

```diff
@@ -29,18 +31,67 @@ export class JournalEffects {
     ofType(journalActions.LOAD_JOURNAL),
     withLatestFrom(
       this.store$.pipe(
-        select(getJournalState)
+        select(getJournalState),
+        map(getLastRefreshed)
+      ),
+      this.store$.pipe(
+        select(getJournalState),
+        map(getSlots)
       )
     ),
-    switchMap(([action, journal]) => {
+    switchMap(([action, lastRefreshed, slots]) => {
       return this.journalProvider
-        .getJournal(journal.lastRefreshed)
+        .getJournal(lastRefreshed)
         .pipe(
-          map((journalData: ExaminerWorkSchedule) => this.slotProvider.detectSlotChanges(journal.slots, journalData)),
+          map((journalData: ExaminerWorkSchedule) => this.slotProvider.detectSlotChanges(slots, journalData)),
           map((slots: any[]) => groupBy(slots, this.slotProvider.getSlotDate)),
           map((slots: {[k: string]: SlotItem[]}) => new journalActions.LoadJournalSuccess(slots)),
           catchError(err => of(new journalActions.LoadJournalFailure(err)))
-        )
-    })
+        );
+    }),
+  )
+
+  @Effect()
+  selectPreviousDayEffect$ = this.actions$.pipe(
+    ofType(journalActions.SELECT_PREVIOUS_DAY),
+    withLatestFrom(
+      this.store$.pipe(
+        select(getJournalState),
+        map(getSelectedDate)
+      ),
+      this.store$.pipe(
+        select(getJournalState),
+        map(canNavigateToPreviousDay)
+      )
+    ),
+    switchMap(([action, selectedDate, canNavigateToPreviousDay]) => {
+      if (!canNavigateToPreviousDay) {
+        return of();
+      }
+      const previousDay = moment(selectedDate).add(-1, 'day').format('YYYY-MM-DD');
+      return of(new journalActions.SetselectedDate(previousDay));
+    }),
+  )
+
+  @Effect()
+  selectNextDayEffect$ = this.actions$.pipe(
+    ofType(journalActions.SELECT_NEXT_DAY),
+    withLatestFrom(
+      this.store$.pipe(
+        select(getJournalState),
+        map(getSelectedDate)
+      ),
+      this.store$.pipe(
+        select(getJournalState),
+        map(canNavigateToNextDay)
+      )
+    ),
+    switchMap(([action, selectedDate, canNavigateToNextDay]) => {
+      if (!canNavigateToNextDay) {
+        return of();
+      }
+      const nextDay = moment(selectedDate).add(1, 'day').format('YYYY-MM-DD');
+      return of(new journalActions.SetselectedDate(nextDay));
+    }),
   )
 }
```

Comment:
Should this do simplistic addition of 1 day / subtraction.... or should it work on stepping to previous date in the array/next date in the array from the state store?

Reply:
We can't guarantee that the `slots` objects contains the slots in such a way that the keys are in chronological order. And the other thing that our action at the moment is to navigate to the next day, not to the next date in the slots object.

There will be a discussion today at some point around skipping days or not skipping days. If we skip days then we need to build it so that we navigate through the array. If we don't skip then we need to leave it as it is.